core/src/cluster_info.rs

```diff
@@ -108,6 +108,8 @@ const GOSSIP_PING_TOKEN_SIZE: usize = 32;
 const GOSSIP_PING_CACHE_CAPACITY: usize = 16384;
 const GOSSIP_PING_CACHE_TTL: Duration = Duration::from_secs(640);
 pub const DEFAULT_CONTACT_DEBUG_INTERVAL: u64 = 10_000;
+/// Limit number of crds values returned when responding to pull-requests.
+const PULL_REQUESTS_OUTPUT_SIZE_LIMIT: usize = 131_072;
```

Comment:
Is this number based on the outbound budget/size of the network?

Reply:
This is based on max `gossip_pull_request-sent_requests` on current testnet and mainnet plus some margin. I realized it is a bit too high, so lowered it to 64k (2^16).

Reply:
ah ok, you think we should scale it based on the number of peers to accommodate network growth?

Reply:
We have a similar case in repair service with a dynamic limit:
https://github.com/solana-labs/solana/blob/5e2d38227fbf91320fcfdfff94b4444cbad3bfff/core/src/serve_repair.rs#L235

Reply:
updated the code to limit number of buffered crds values based on outbound data budget.